### PR TITLE
refactor authentication resource management

### PR DESCRIPTION
**Pull request by rupansh-gob**
this refactors how we manage authentication related resources

This should decrease the complexity of using authenticated canisters object
Should prevent hydration bugs (and fix the existing ones), and stop headaches while development e.g components disappearing for no apparent reason inside `<Suspense>`

Staging: https://hot-or-not-web-leptos-ssr-staging.fly.dev/

### CODE REVIEW

File: ssr/src/auth/mod.rs [hunk 1]
```diff
@@ -44,15 +44,21 @@ impl DelegatedIdentityWire {
...
         };
 
-        let mut delegation_chain = from.delegation_chain();
-        delegation_chain.push(signed_delegation);
-
         Self {
             from_key: sig.public_key.unwrap(),
             to_secret: to_secret.to_jwk(),
-            delegation_chain,
+            delegation_chain: vec![signed_delegation],
```
Comment (ravibazz) [inline ssr/src/auth/mod.rs:50]: Shouldn't this also include signed_delegation of `from_identity` (there could be a case where from is a delegated Identity)

Comment (ravibazz) [reply ssr/src/auth/mod.rs:50]: We create delegation for video importer from delegated Identity.

File: ssr/src/component/content_upload.rs [hunk 2]
```diff
@@ -30,50 +23,71 @@ pub fn YoutubeUpload(
                 .upload_content(url_value(), delegated_identity)
                 .await;
             match res {
-                Err(e) => response.set(e.to_string()),
-                _ => response.set("Submitted!".to_string()),
```
[the remark this answers is not in the document]
Comment (rupansh-gob) [reply ssr/src/component/content_upload.rs:34]: https://github.com/go-bazzinga/hot-or-not-web-leptos-ssr/pull/320/files/71f2cb438ad391d686df004f6702e5485e279130#diff-c7f40b5b72355aa476257435c6c2a63808aa8d5cf7e8cda4f9ff8c211caa6650R63

[the remark this answers is not in the document]
Comment (ravibazz) [reply ssr/src/component/content_upload.rs:34]: The value is not getting returned. I think the semi-colon on match block is not required

**Review by ravibazz — APPROVED** [review]

**Pull request merged**